.circleci/config.yml

```diff
@@ -48,6 +48,7 @@ commands:
     description: 'Running linters'
     steps:
       - run: yarn run jest --projects jest.eslint.config.js jest.stylelint.config.js --maxWorkers=3 --reporters jest-silent-reporter
+      - run: yarn typecheck
```

Comment:
So, if the typings update, does it also commit the changes to the file? 😕

Reply:
I don't understand your question. Are you sure you understand what this tool does?

Reply:
What does this tool do then? Does this PR not intend to deliver https://github.com/commercetools/ui-kit/issues/853?

Reply:
No this is a different approach from the prop-types. Probably it won't really work because we need to have typescript files in order for the compiler to extract the declarations, even if we infer them from the prop-types.

This approach though requires some manual work in terms of maintenance but allows to expose declarations without having to setup typescript and migrate stuff.
The `dtslint` tool allows to "check" the type declarations that we define if it's valid syntax or not. It obviously does not guarantee that the types are "in sync" with the components but it's a good start.
This is also what you would use to publish type declarations of a JS library to [definitely typed](https://github.com/DefinitelyTyped/DefinitelyTyped).

Reply:
Ah okay, so to clarify.

Step 1, you write your Typescript types for your component
Step 2, you run `dtslint` to check those values.

Okay yeah I get it now.

But now we need to maintain our typings, our readmes, and our proptypes, all independently. I don't really see how it helps us that much.

Reply:
Well, eventually we would probably want to only have the types, so we might need to migrate the codebase to TS.

The readme stuff I think is a general problem regardless if we use TS or not. Maybe we could also write our own docs parser if we don't find the right tool.

Reply:
> eventually we would probably want to only have the types, so we might need to migrate the codebase to TS

Like I have mentioned before, I would like to triage our UI-Kit infrastructure refactorings, and prioritize them. Justifying the TS migration this way feels a bit like jumping the queue...

Reply:
Since we can't generate the typings, manually making them seems like too much work to do without including our PO and defining the task.

Reply:
Sorry but no one said to migrate anything nor to prioritize this. I'm not sure where did you get that impression.

That's why I looked into this approach which is very lightweight and does not involve any change in the existing "infrastructure". It's also marked as "experiment" to emphasis that.

Reply:
Alright, so to confirm, you don't plan on manually adding types for all of the components?